docs/promtail-failure-modes.md

```diff
@@ -0,0 +1,57 @@
+# Promtail Failure Modes
+
+This document describes known failure modes of `promtail` on edge cases and the adopted trade-offs.
+
+
+## A tailed file is truncated while `promtail` is not running
+
+Given the following order of events:
+
+1. `promtail` is tailing `/app.log`
+2. `promtail` current position for `/app.log` is `100` (bytes)
+3. `promtail` is stopped
+4. `/app.log` is truncated and new logs are appended to it
+5. `promtail` is restarted
+
+When `promtail` is restarted, it reads the previous position (`100`) from the positions file. Two scenarios are then possible:
+
+- `/app.log` size is < than the position before truncating
+- `/app.log` size is >= than the position before truncating
+
+If the `/app.log` file size is less than the previous position, then the file is detected as truncated and logs will be tailed starting from position `0`. Otherwise, if the `/app.log` file size is >= than the previous position, `promtail` can't detect it was truncated while not running and will continue tailing the file from position `100`.
+
```

Comment:
Might be worth noting if the log file rolled mulitple times while promtail wasn't running and the size is greater than the position from the file it will start at the position from the file.  Or put another way, promtail does not do anything fancy like track a hash of the log file to know if it's actually continuing from the same file or not

Reply:
Somewhere I feel like we need to document the advantages of using larger log files in regards to promtail and decreasing odds of lost log files.  The less frequently log files are rolled the better success promtail has when things go wrong... Not sure where this might go but maybe it fits on this page?